homeassistant/components/ezviz/config_flow.py

```diff
@@ -0,0 +1,252 @@
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import AbortFlow
+
+from .const import (  # pylint: disable=unused-import
+    ATTR_SERIAL,
+    ATTR_TYPE_CAMERA,
+    ATTR_TYPE_CLOUD,
+    CONF_FFMPEG_ARGUMENTS,
+    DEFAULT_CAMERA_USERNAME,
+    DEFAULT_FFMPEG_ARGUMENTS,
+    DEFAULT_REGION,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EzvizConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Ezviz."""
+
+    VERSION = 1
+    CONNECTION_CLASS = CONN_CLASS_CLOUD_POLL
+
+    async def _validate_and_create_auth(self, data):
+        """Try to login to ezviz cloud account and create entry if successful."""
+        await self.async_set_unique_id(data[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        client = EzvizClient(
+            data[CONF_USERNAME],
+            data[CONF_PASSWORD],
+            data.get(CONF_REGION, DEFAULT_REGION),
+            data.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+        )
+
+        # Attempts a login request.
+        try:
+            await self.hass.async_add_executor_job(client.login)
+
+        except (PyEzvizError, HTTPError) as err:
+            raise InvalidAuth from err
+
+        auth_data = {
+            CONF_USERNAME: data[CONF_USERNAME],
+            CONF_PASSWORD: data[CONF_PASSWORD],
+            CONF_REGION: data.get(CONF_REGION, DEFAULT_REGION),
+            CONF_TYPE: ATTR_TYPE_CLOUD,
+        }
+
+        return self.async_create_entry(title=data[CONF_USERNAME], data=auth_data)
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return EzvizOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        # Check if ezviz cloud account is present in entry config.
+        # Return camera user flow if present.
+        for item in self._async_current_entries():
+            if item.data.get(CONF_TYPE) == ATTR_TYPE_CLOUD:
+                return await self.async_step_user_camera()
+
+        errors = {}
+
+        if user_input is not None:
+            if CONF_TIMEOUT not in user_input:
+                user_input[CONF_TIMEOUT] = DEFAULT_TIMEOUT
+
+            try:
+                return await self._validate_and_create_auth(user_input)
+
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+            except AbortFlow:
+                raise
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): str,
+                vol.Required(CONF_PASSWORD): str,
+                vol.Optional(CONF_REGION, default=DEFAULT_REGION): str,
```

Comment:
We changed to url it seems.

How will the user know the server url?

Reply:
It looks like EZVIZ has 3x isolated regions. The default url will work for 95% of the globe. So far only Russia users will need to change the url.

Would it be possible to use a menu with a "custom" option for someone to type their own url?

Reply:
We could use `vol.In(regions + ["custom"])` and let the `"custom"` choice lead to a separate step where the user can enter a url.